hidimstat/estimator/tests/test_Dnn_learner_single.py

```diff
@@ -0,0 +1,103 @@
+    assert not np.all(predict_prob == predict_prob[0])
+    # Check if the predicted probabilities are not all zeros
+    assert not np.all(predict_prob == 0)
+    # Check if the predicted probabilities are not all ones
+    assert not np.all(predict_prob == 1)
+    # Check if the predicted probabilities are not all the same for each class
+    assert not np.all(predict_prob[:, 0] == predict_prob[0, 0])
+    assert not np.all(predict_prob[:, 1] == predict_prob[0, 1])
+    # Check if the predicted probabilities are not all zeros for each class
+    assert not np.all(predict_prob[:, 0] == 0)
+    assert not np.all(predict_prob[:, 1] == 0)
+    # Check if the predicted probabilities are not all ones for each class
+    assert not np.all(predict_prob[:, 0] == 1)
+    
+
+def test_DNN_single_binary():
+    """
+    Test the DNN learner single for binary classification.
+    """
+    X, y = generate_data(problem_type="classification")
+    learner = DnnLearnerSingle(do_hypertuning=True, problem_type="binary", n_jobs=10, verbose=0)
+    learner.fit(X, y)
+    predict_prob = learner.predict_proba(X)
+    # Check if the predicted probabilities sum up to 1 for each instance
+    assert np.allclose(np.sum(predict_prob, axis=1), 1)
+    # Check if the predicted class labels match the true labels for at least one instance
+    assert np.sum(np.argmax(predict_prob, axis=1) == y) > 0
+    assert np.all(np.max(predict_prob, axis=1) >= 0.5)
+    assert np.all(np.min(predict_prob, axis=1) < 0.5)
+    # Check if the maximum predicted probability is greater than 0.95
+    assert 0.95 < np.max(predict_prob)
+    # Check if the minimum predicted probability is less than 0.05
+    assert 0.05 > np.min(predict_prob)
+    # Check if the predicted probabilities are not all the same
+    assert not np.all(predict_prob == predict_prob[0])
+    # Check if the predicted probabilities are not all zeros
+    assert not np.all(predict_prob == 0)
+    # Check if the predicted probabilities are not all ones
+    assert not np.all(predict_prob == 1)
+    # Check if the predicted probabilities are not all the same for each class
+    assert not np.all(predict_prob[:, 0] == predict_prob[0, 0])
+    assert not np.all(predict_prob[:, 1] == predict_prob[0, 1])
+    # Check if the predicted probabilities are not all zeros for each class
+    assert not np.all(predict_prob[:, 0] == 0)
+    assert not np.all(predict_prob[:, 1] == 0)
+    # Check if the predicted probabilities are not all ones for each class
+    assert not np.all(predict_prob[:, 0] == 1)
+
+
+def test_DNN_single_ordinal():
+    """
+    Test the DNN learner single for ordinal.
+    """
+    X, y = generate_data(problem_type="classification")
+    learner = DnnLearnerSingle(do_hypertuning=True, problem_type="ordinal", n_jobs=10, verbose=0)
+    learner.fit(X, y)
+    predict_prob = learner.predict_proba(X)[:,0]
+    # Check if the predicted class labels match the true labels for at least one instance
```

Comment:
Need an assert...

Reply:
This will be addressed in the future (see issue #79).